src/Controller.php

```diff
@@ -63,17 +70,100 @@ public function connect(Application $app)
      */
     public function thumbnail(Application $app, Request $request, $file, $action, $width, $height)
     {
+        // Set to default 404 image if restricted to aliases
+        if($this->isRestricted($app, $request)) {
+            $transaction = $this->defaultTransaction($app, $request);
+            $thumbnail = $app['thumbnails']->respond($transaction);
+            return new Response($thumbnail);
```

Comment:
`defaultTransaction` could be changed to `defaultResponse` to dedup here and below.